website/server/controllers/api-v3/auth.js

```diff
@@ -312,10 +314,30 @@ api.pusherAuth = {
     let socketId = req.body.socket_id;
     let channelName = req.body.channel_name;
 
-    let isPresenceChannel = channelName.indexOf('presence-') === 0;
+    // Channel names are in the form of {presence|private}-{group|...}-{resourceId}
+    let [channelType, resourceType, ...resourceId] = channelName.split('-');
+
+    if (PUSHER_CHANNEL_TYPES.indexOf(channelType) === -1) {
+      throw new BadRequest('Invalid Pusher channel type.');
+    }
+
+    if (resourceType !== 'group') { // only groups are supported
+      throw new BadRequest('Invalid Pusher resource type.');
+    }
+
+    resourceId = resourceId.join('-'); // the split at the beginning had split resourceId too
+    if (!validator.isUUID(resourceId)) {
+      throw new BadRequest('Invalid Pusher resource id, must be a UUID.');
```

Comment:
These all look like implementation details, so as long as the API is private, I think it's fine to not localize them. A user should never see these errors.